src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -444,6 +490,25 @@ private static boolean hasLineBreakBefore(DetailAST rightCurly) {
     }
 
     /**
+     * Checks if right curly is alone on line in token the lambda.
+     * @param details for validation.
+     * @return true, if right curly is alone on line.
+     */
+    private static boolean isAloneInLambda(Details details) {
+        final DetailAST lcurly = details.lcurly;
+        final DetailAST rcurly = details.rcurly;
+        final int lcurlyNo = lcurly.getLineNo();
+        final int rcurlyNo = rcurly.getLineNo();
+        if (rcurly.getPreviousSibling() == null) {
+            return rcurlyNo != lcurlyNo;
+        }
+        else {
+            return rcurlyNo != lcurlyNo
+                && rcurlyNo != rcurly.getPreviousSibling().getLineNo();
```

Comment:
I did two return by reason of NPE in empty lambda () -> {}. Now is ok in lambda of such a type () -> {}, such lambda are many in openjdk. But in lambda of such a type () -> {
} check is fail.  Maybe, will find such lambda chance very small and skip this fail.